### PR TITLE
Task 1 Using UNet to Segment the ISIC 2017/8 dataset Raul Movila S4703527

### PR DESCRIPTION
**Pull request by mraula**
Hello,

The pull request is for task 1 of the Pattern Analysis 2023 report

The mean val_dice_coef after training resulted to 0.83335. 

The pull request contains the following files:

```sh
1. `dataset.py`: # Loads and pre-processes the dataset, including the training, validation, and testing sets.
2. `model.py`: # Defines the UNet architecture, including encoder and decoder blocks, Return the model.
3. `train.py`: # Implements the training procedure for the model it trains the model then saves the best model.
4. `predict.py`: # Predict mask based on saved UNet model, performing predictions, and calculating evaluation metrics.
5. `utils.py`: # Provides utility functions used throughout the project
6. 'files' : # Stores all the data from training
7. 'results' : # Stores all the resulted masks
```

Refer to README for further information.

Thank you,
Raul Andrei Movila S4703527

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Well organized files.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.83. Predicted Mask output looks good. 
- Try to use TensorFlow to construct the Improved U-Net model, but the model body is a U-Net model(filters start with 64 and end up with 1024) instead of the improved U-Net one(filters start with 16 and end up with 256). No data augmentation.
- Sufficient docstring and comment in the code, but need some references.
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, but not correct model, not improved UNet -4
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, but could have more info -1
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove the results data images, repos should not have data in it -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.

**Comment by mraula** [comment]
Feedback Implemented (removed data from repo).

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.

**Comment by shakes76** [comment]
Unfortunately will not be merged because once data files are added, its in the repo forever (even if removed, its in history). Only affects merge though, grading is complete, no marks will be gained or lost. Feedback marks already gained as commented above.
